kvbc/src/categorization/blocks.cpp

```diff
@@ -41,31 +43,21 @@ BlockMerkleInput RawBlock::getUpdates(const std::string& category_id,
                                       const BlockMerkleOutput& update_info,
                                       const BlockId& block_id,
                                       const std::shared_ptr<storage::rocksdb::NativeClient>& native_client) {
```

Comment:
I was wondering if it makes sense to make the categories (and the native client for that matter) RawBlock members. That way we can avoid ref count updates in the categories and simplify the interface a bit.

Reply:
I took your advice but didn't make it as a member in order to keep the stateless property and avoid coping.
Instead, I pass the categories as const pointer (since I need also a default constructor).
Please review it